plugins/auth-node/api-report.md

```diff
@@ -28,12 +28,12 @@ export class DefaultIdentityClient implements IdentityApi {
   // @deprecated
   authenticate(token: string | undefined): Promise<BackstageIdentityResponse>;
   static create(options: IdentityClientOptions): DefaultIdentityClient;
-  // (undocumented)
-  getIdentity({
-    request,
-  }: IdentityApiGetIdentityRequest): Promise<
-    BackstageIdentityResponse | undefined
-  >;
+  getIdentity(
+    options: IdentityApiGetIdentityRequest,
+  ): Promise<BackstageIdentityResponse | undefined>;
+  getUserIdentity(
+    options: IdentityApiGetIdentityRequest,
+  ): Promise<UserTokenIdentity | undefined>;
```

Comment:
We discussed this a bit in person (@freben and I), and feeling that this is perhaps a bit too early and needs a clearer view of the overall direction of backend auth and identities. There's a pretty big risk that we're headed towards a local maximum here, rather than something that takes the entire system into consideration.

For example, one reasonable alternative would be that we go for separate `UserEntityService` and `ServiceIdentityService`. It's going to depend quite a lot on how plugins are supposed to interact with service identities and what kind of different service identities that we end up having. Seeing as we do already have the token manager and identity API, it feels like that is an alright state to be in for now and we don't particularly have anything to gain from doing small evolutions of the APIs just yet.